troposphere/static/js/components/common/SSHCreateView.jsx

```diff
@@ -0,0 +1,179 @@
+import React from "react";
+import RaisedButton from "material-ui/RaisedButton";
+
+export default React.createClass({
+    displayName: "SSHCreateView",
+
+    getInitialState: function() {
+        return {
+            keyName: "",
+            publicKey: "",
+            showValidation: false
+        };
+    },
+
+    validateName: function () {
+        let name = this.state.keyName;
+        let hasError = false;
+        let message = "";
+        let maxCharLen = 60;
+
+        if (name === "") {
+            hasError = true;
+            message = "This field is required";
+        }
+
+        if (name.length > maxCharLen) {
+            hasError = true;
+            message = `Must be less than ${maxCharLen} characters long`;
+        }
+
+        return {
+            hasError,
+            message
+        }
+    },
+
+    validatePublicKey: function () {
+        let pubKey = this.state.publicKey;
+        let parts = pubKey.split(/\s+/g)
+        let hasError = false;
+        let message = "";
+
+
+        if (pubKey === "") {
+            hasError = true;
+            message = "This field is required";
+        }
+
+        if (!this.validateKeyType(parts[0])) {
+            hasError = true;
+            message = "Public key must begin with either ssh-rsa, ssh-dss, ecdsa-sha2-nistp256, or ssh-ed25519";
+        }
+
+        return {
+            hasError,
+            message
+        }
+    },
+
+    validateKeyType: function(firstWord) {
+        return /^(ssh-dss|ecdsa-sha2-nistp256|ssh-ed25519|ssh-rsa)$/.test(firstWord);
+    },
+
+    isSubmittable: function () {
+        if (!this.validateName().hasError && !this.validatePublicKey().hasError) {
```

Comment:
I dont know javascript, but can't you directly return the output of the check here?

`return(!this.validateName().hasError && !this.validatePublicKey().hasError)`

Reply:
I think it depends on the JS standards, for example, ES5 and ES6. 

We may can do that, but I am trying to follow the standards that the current codebase follows :)

I will note this down and revisit again later. And potentially, a PR for cleanup

Thanks!!